packages/donate-button/src/generic-button-loader.tsx

```diff
@@ -0,0 +1,54 @@
+import {render} from 'preact';
+import {
+	GenericButtonProps,
+	DonateButtonOptions
+} from 'src/helpers/options-types';
+
+interface GenericButtonLoader {
+	selector: string;
+	onClick: () => void;
+	widgetOptions: Partial<DonateButtonOptions>;
+	options: Partial<GenericButtonProps>;
+}
+
+const getCharityName = (
+	widgetOptions: Partial<DonateButtonOptions>
+): string => {
+	const genericFoundation = 'your-foundation';
+	if (typeof widgetOptions.onSubmit === 'function') {
+		return genericFoundation;
+	}
+
+	return widgetOptions.onSubmit?.charity
+		? widgetOptions.onSubmit?.charity
+		: genericFoundation;
+};
+
+const genericButtonLoader = ({
+	selector,
+	options = {},
+	onClick,
+	widgetOptions
+}: GenericButtonLoader) => {
+	const div = document.querySelector(selector);
+
+	const charity = getCharityName(widgetOptions);
+	const hrefUrl = `https://www.every.org/${charity}/donate`;
+
+	if (div) {
+		import('./components/GenericButton')
```

Comment:
Given that this is the default i'm OK with not putting it behind a loader and just always loading it. It's negligible bundle size and would result in a faster experience to not have to chain HTTP calls.